src/controller/rbac-controller.ts

```diff
@@ -42,28 +47,60 @@ export default class RbacController extends BaseController {
       '/roles': {
         GET: {
           policy: async () => true,
-          handler: this.returnAllRoles.bind(this),
+          handler: this.getAllRoles.bind(this),
+        },
+        POST: {
+          policy: async (req) => this.roleManager.can(req.token.roles, 'create', 'all', 'Role', ['*']),
+          handler: this.createRole.bind(this),
+          body: { modelName: 'UpdateRoleParams' },
+        },
+      },
+      '/roles/:id(\\d+)': {
+        GET: {
+          policy: async () => true,
+          handler: this.getSingleRole.bind(this),
+        },
+        PATCH: {
+          policy: async (req) => this.roleManager.can(req.token.roles, 'update', 'all', 'Role', ['*']),
+          handler: this.updateRole.bind(this),
+          body: { modelName: 'UpdateRoleParams' },
+        },
+        DELETE: {
+          policy: async (req) => this.roleManager.can(req.token.roles, 'delete', 'all', 'Role', ['*']),
+          handler: this.deleteRole.bind(this),
+        },
+      },
+      '/roles/:id(\\d+)/permissions': {
+        POST: {
```

Comment:
You can add permissions in bulk (as the POST request takes an array of permission rules), but you can only delete them one at a time. This is because - with a POST request - it is easy to allow an array of objects as a request body, but DELETE requests do not have such a body, so you need to put everything in the URL.